bdk_core/src/coin_select.rs

```diff
@@ -196,10 +272,27 @@ impl CoinSelector {
             }
         };
 
-        let (total_weight, fee) = if use_drain {
-            (weight_with_drain, target_fee_with_drain)
+        let (total_weight, fee, long_term_fee) = if use_drain {
+            (
+                weight_with_drain,
+                target_fee_with_drain,
+                long_term_fee_with_drain,
+            )
         } else {
-            (base_weight, target_fee_without_drain)
+            (
+                base_weight,
+                target_fee_without_drain,
+                long_term_free_without_drain,
+            )
+        };
+
+        let waste = {
+            let base_waste = if use_drain {
+                self.opts.drain_cost()
+            } else {
+                excess
+            };
+            (base_waste + fee) as i64 - long_term_fee as i64
```

Comment:
why does the waste go down if the `long_term_fee` goes up??

Reply:
I'm sorry, do you mind elaborating?

Reply:
the higher your `long_term_fee` is the more waste there is no? A tx that creates a high long term fee is more wasteful than one that doesn't. If this is correct then this contradicts the calculation here since if `long_term_fee` is higher `waste` is lower. It's possible I am completely misunderstanding the concept here.

Reply:
`waste` should be the waste of not spending it now.

Reply:
Ok I think I get it now. I had a different idea in mind. Might be worth a comment to explain the concepts.